### ISSUE TITLE
urlForResource() returns wrong URL for nested resources (eg, /foo/bar/baz)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1874]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#1874|
|Type|defect|
|Created|2006-06-28 10:44:34Z|

When looking up a resource that's a couple of levels down in the URL namespace, urlForResource() fails to work.  Looks like _handleSegment is registering each segment rather than the full path, which is not nice at all.

<details><summary>Searchable metadata</summary>

```
trac-id__1874 1874
type__defect defect
reporter__wsanchez wsanchez
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__dav dav
time__1151491474000000 1151491474000000
changetime__1151671554000000 1151671554000000
version__None None
owner__wsanchez wsanchez
cc__wsanchez
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez commented|
|-|-|

Fixed on branch: urlForResource-1874

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez commented|
|-|-|

Passed on buildbot: full-2.3, OS X, win32er

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @glyph|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @wsanchez|
|-|-|


Fix before merging:

 * remove spurious whitespace added in ```test_server.py``` line 357.
 * neither ```test_requestedResource``` and ```test_deepResource``` has a docstring explaining what is being tested.

Other:

 * _handleSegment could stand to be a bit better documented; it's pretty complex, but there's no description of its inputs or outputs.

Otherwise looks good.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set status to `closed`|
|-|-|

(In [17431]) Fix urlForResource() for nested resources.
Author: wsanchez
Reviewed by glyph
Fixes [#1874](https://github.com/twisted/twisted/issues/1874)
